ci-operator/step-registry/openstack/gather/openstack-gather-commands.sh

```diff
@@ -17,24 +25,34 @@ collect_bootstrap_logs() {
 			echo "Collecting bootstrap logs..."
 			CLUSTER_ID=${BOOTSTRAP_NODE%-bootstrap}
 			openstack security group rule create ${CLUSTER_ID}-master --protocol tcp --dst-port 22:22 --remote-ip 0.0.0.0/0
-			FIP=$(openstack floating ip create "$OPENSTACK_EXTERNAL_NETWORK" --description "${CLUSTER_ID}-bootstrap" --format value --column floating_ip_address)
-			FIPS+=("${FIP}")
-			openstack server add floating ip ${CLUSTER_ID}-bootstrap ${FIP}
-			GATHER_BOOTSTRAP_ARGS+=('--bootstrap' "${FIP}")
+			if [[ $CREATE_FIPS == 1 ]]; then
+				IP=$(openstack floating ip create "$OPENSTACK_EXTERNAL_NETWORK" --description "${CLUSTER_ID}-bootstrap" --format value --column floating_ip_address)
+				FIPS+=("${IP}")
+				openstack server add floating ip ${CLUSTER_ID}-bootstrap ${IP}
+			else
+				IP=openstack server show ${BOOTSTRAP_NODE} --colum addresses --format json | jq -r '.addresses."'${CLUSTER_ID}'-openshift"[0]'
+			fi
+			GATHER_BOOTSTRAP_ARGS+=('--bootstrap' "${IP}")
 			
 			for idx in 0 1 2; do
 				if openstack server show ${CLUSTER_ID}-master-${idx} &> /dev/null; then
-					FIP=$(openstack floating ip create "$OPENSTACK_EXTERNAL_NETWORK" --description "${CLUSTER_ID}-master-${idx}" --format value --column floating_ip_address)
-					FIPS+=("${FIP}")
-					openstack server add floating ip ${CLUSTER_ID}-master-${idx} ${FIP}
-					GATHER_BOOTSTRAP_ARGS+=('--master' "${FIP}")
+					if [[ $CREATE_FIPS == 1 ]]; then
+						IP=$(openstack floating ip create "$OPENSTACK_EXTERNAL_NETWORK" --description "${CLUSTER_ID}-master-${idx}" --format value --column floating_ip_address)
+						FIPS+=("${IP}")
+						openstack server add floating ip ${CLUSTER_ID}-master-${idx} ${IP}
+					else
+						IP=openstack server show ${CLUSTER_ID}-master-${idx} --colum addresses --format json | jq -r '.addresses."'${CLUSTER_ID}'-openshift"[0]'
+					fi
+					GATHER_BOOTSTRAP_ARGS+=('--master' "${IP}")
 				fi
 			done
 			
 			SSH_PRIV_KEY_PATH="${CLUSTER_PROFILE_DIR}/ssh-privatekey"
 			openshift-install gather bootstrap --key "${SSH_PRIV_KEY_PATH}" "${GATHER_BOOTSTRAP_ARGS[@]}"
 			cp log-bundle-*.tar.gz "${ARTIFACT_DIR}"
-			openstack floating ip delete "${FIPS[@]}"
+			if [[ ${#FIPS[@]} -ge 0 ]]; then
```

Comment:
you probably want `gt` (Greater Than) rather than `ge` (Greater than or Equal to) here